packages/blob/src/index.js

```diff
@@ -1,22 +1,30 @@
 /**
  * Browser dependencies
  */
-const { fetch } = window;
+const { fetch, File } = window;
 const { createObjectURL, revokeObjectURL } = window.URL;
 
 const cache = {};
 
 export function createBlobURL( blob ) {
+	const { name, lastModified, type } = blob;
 	const url = createObjectURL( blob );
 
-	cache[ url ] = blob;
+	cache[ url ] = {
+		blob,
+		name,
+		lastModified,
+		type,
+	};
 
 	return url;
 }
 
 export function getBlobByURL( url ) {
 	if ( cache[ url ] ) {
-		return Promise.resolve( cache[ url ] );
+		const { blob, name, lastModified, type } = cache[ url ];
+		const file = new File( [ blob ], name, { lastModified, type } );
```

Comment:
Would caching new File( [ blob ], name, { lastModified, type } ); directly in cache[ url ] be an option?

Reply:
e.g. in createBlobURL do cache[ url ] = new File( [ blob ], name, { lastModified, type } );

Reply:
Sure, sounds good